imap_processing/spice/repoint.py

```diff
@@ -1,16 +1,49 @@
 """Functions for retrieving repointing table data."""
 
 import logging
-import os
 from pathlib import Path
 from typing import Union
 
 import numpy as np
 import pandas as pd
 from numpy import typing as npt
 
+# Copy-on-write will be enabled by default in pandas 3.0
+# It is recommended to enable it now. See:
+# https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
+pd.options.mode.copy_on_write = True
+
 logger = logging.getLogger(__name__)
 
+# Use a mutable module level attribute to store the location of the repoint table
+_repoint_table_path: Path | None = None
+# cache repoint-table data to avoid reloading from disk every time
+_repoint_df_cache: dict[str, pd.DataFrame] = {}
+
+
+def set_repoint_table_paths(paths: list[Path]) -> None:
```

Comment:
Do we want to name these similar to SPICE so call it furnish_repoint_file() or something like that? Maybe set_global_repoint_table() to add in global and indicate it is updating this globally.